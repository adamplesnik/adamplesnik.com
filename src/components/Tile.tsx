import { clsx } from 'clsx'
import { MousePointerClick } from 'lucide-react'
import { HTMLAttributes, ReactNode } from 'react'
import Heading from './Heading'
import Icon from './Icon'

const Tile = ({ sub, children, title, badge, links, className }: TileWrapperProps) => {
  return (
    <div className={clsx('px-12', className)}>
      <div className="flex gap-1">
        {title && <Heading size={sub ? 3 : 2}>{title}</Heading>}
        {badge && badge}
      </div>
      {children}
      {links && (
        <div className="flex items-baseline gap-2">
          <Icon Icon={MousePointerClick} />
          <div className="flex flex-col gap-1">{links}</div>
        </div>
      )}
    </div>
  )
}

type TileWrapperProps = {
  badge?: ReactNode
  links?: ReactNode
  sub?: boolean
  title?: string
} & HTMLAttributes<HTMLDivElement>

export default Tile
