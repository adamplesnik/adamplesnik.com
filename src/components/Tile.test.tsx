import { toJson } from '@/utils/toJson'
import renderer from 'react-test-renderer'
import { expect, test } from 'vitest'
import Tile from './Tile'

test('Tile', () => {
  const component = renderer.create(<Tile title="Test Title">Tile</Tile>)
  const tree = toJson(component)
  expect(tree).toMatchSnapshot()
})

test('Tile with badge', () => {
  const component = renderer.create(
    <Tile title="Test Title" badge={<span />}>
      Tile with badge
    </Tile>
  )
  const tree = toJson(component)
  expect(tree).toMatchSnapshot()
})
