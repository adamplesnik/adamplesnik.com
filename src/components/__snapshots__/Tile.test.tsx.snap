// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tile 1`] = `
<div
  className="overflow-hidden rounded-lg border border-zinc-600/20 bg-gradient-to-tr from-zinc-800/20 to-zinc-800/40 px-6 pb-4 pt-8 shadow-2xl"
>
  <h3
    className="relative w-full font-medium pb-4 text-xl text-slate-200"
  >
    Test Title
  </h3>
  <h4
    className="relative w-full font-medium pb-6 text-sm opacity-70"
  >
    Test Subtitle
  </h4>
  Tile
</div>
`;

exports[`Tile with class name 1`] = `
<div
  className="overflow-hidden rounded-lg border border-zinc-600/20 bg-gradient-to-tr from-zinc-800/20 to-zinc-800/40 px-6 pb-4 pt-8 shadow-2xl test-class"
>
  <h3
    className="relative w-full font-medium pb-4 text-xl text-slate-200"
  >
    Test Title
  </h3>
  <h4
    className="relative w-full font-medium pb-6 text-sm opacity-70"
  >
    Test Subtitle
  </h4>
  Tile with class name
</div>
`;
